### PR TITLE
C++ Script fix ref Fumping and The Big Bone Worm #11418

### PR DESCRIPTION
**Pull request by ex1dreamer**
**Changes proposed:**

-  Updated two C++ scripts related to Fumping and The Big Bone Worm #11418

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5

**Issues addressed:**

Closes #  (11418)

**Tests performed:**

(Does it build, tested in-game, etc.)
Tested in game

**Known issues and TODO list:** (add/remove lines as needed)

- update behaviour of summoned creatures
-  update SQL table spell_script_names -> spell D=39246, script name=spell_q10930_big_bone_worm

### CODE REVIEW

File: src/server/scripts/Spells/spell_quest.cpp [hunk 1]
```diff
@@ -2643,43 +2643,90 @@ class spell_q14100_q14111_make_player_destroy_totems : public SpellScriptLoader
...
+    void Register() override
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_q10929_fumping::SetDest, EFFECT_1, TARGET_DEST_CASTER);
+    }
+};
 
-            void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
-            {
-                if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
-                    return;
```
Comment (Aokromes) [inline src/server/scripts/Spells/spell_quest.cpp:2670]: remove this line.

**Comment by jackpoz** [comment]
Could you please describe more in detail what and how this pr changes ?

**Comment by ex1dreamer** [comment]
> Could you please describe more in detail what and how this pr changes ?

1. Changes existing script for spell 39238 - Fumping. Mainly to align with summon with retail behaviour. It will spawn  randomly one sand gnome (server side spell 39240); 3 sand gnomes (server side spell 39247) or one matur bone sifter (server side spell 39241)
2.  creates missing script for spell 39246 (quest ID10930 big bone worm). The sript will be activated on condition of dead corpse of Decrepit Clefthoof (creature ID 22105) in  5 yards range. It will spawn on randomly one mature bone sifter (server side spell 39241), 3 mature bone sifters (server side spell 39245) or, finally, elite mob Haishlud (server side spell 39248)
3. Small fix  to show aura corrently for 39238 Fumping.
4.  Tested in game, C++ works. Smart scipts in the database for summoned creatures still do not work correctly.

File: src/server/scripts/Spells/spell_quest.cpp [hunk 2]
```diff
@@ -2643,43 +2643,90 @@ class spell_q14100_q14111_make_player_destroy_totems : public SpellScriptLoader
...
+    PrepareAuraScript(spell_q10929_fumpingAuraScript);
 
-        void Register() override
-        {
-            OnEffectRemove += AuraEffectRemoveFn(spell_q10929_fumpingAuraScript::HandleEffectRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
-        }
-    };
+    bool Validate(SpellInfo const* /*spell*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_SAND_GNOME,SPELL_SUMMON_SAND_GNOME3, SPELL_SUMMON_MATURE_BONE_SIFTER });//changed 24042021
```
Comment (jackpoz) [inline src/server/scripts/Spells/spell_quest.cpp:2677]: we usually don't include comments about when something was changed, there is `git blame` for that

File: src/server/scripts/Spells/spell_quest.cpp [hunk 3]
```diff
@@ -2643,43 +2643,90 @@ class spell_q14100_q14111_make_player_destroy_totems : public SpellScriptLoader
...
+    }
 
-    AuraScript* GetAuraScript() const override
+    void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
+            return;
+
+        if (Unit* caster = GetCaster())
+            caster->CastSpell(caster, RAND(SPELL_SUMMON_SAND_GNOME, SPELL_SUMMON_SAND_GNOME3, SPELL_SUMMON_MATURE_BONE_SIFTER), true);//changed 24042021
```
Comment (jackpoz) [inline src/server/scripts/Spells/spell_quest.cpp:2686]: same here about comments with modified date

File: src/server/scripts/Spells/spell_quest.cpp [hunk 4]
```diff
@@ -2643,43 +2643,90 @@ class spell_q14100_q14111_make_player_destroy_totems : public SpellScriptLoader
...
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_q10930_big_bone_worm::SetDest, EFFECT_1, TARGET_DEST_CASTER);
+    }
+};
+
+class spell_q10930_big_bone_worm_AuraScript : public AuraScript
+{
+    PrepareAuraScript(spell_q10930_big_bone_worm_AuraScript);
+    bool Validate(SpellInfo const* /*spell*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_MATURE_BONE_SIFTER,SPELL_SUMMON_MATURE_BONE_SIFTER3, SPELL_SUMMON_HAISHULUD });//changed 24042021
```
Comment (jackpoz) [inline src/server/scripts/Spells/spell_quest.cpp:2717]: same here about comments with modified date

**Review by jackpoz — CHANGES_REQUESTED** [review]
I left some line comments

**Comment by jackpoz** [comment]
I changed the status of this PR to Draft as the SQL bit is missing, so it's clear not to merge it yet.

File: src/server/scripts/Spells/spell_quest.cpp (hunk 4 above)
Comment (Faq) [reply src/server/scripts/Spells/spell_quest.cpp:2717]:
```suggestion
        return ValidateSpellInfo({ SPELL_SUMMON_MATURE_BONE_SIFTER,SPELL_SUMMON_MATURE_BONE_SIFTER3, SPELL_SUMMON_HAISHULUD });
```

File: src/server/scripts/Spells/spell_quest.cpp (hunk 3 above)
Comment (Faq) [reply src/server/scripts/Spells/spell_quest.cpp:2686]:
```suggestion
            caster->CastSpell(caster, RAND(SPELL_SUMMON_SAND_GNOME, SPELL_SUMMON_SAND_GNOME3, SPELL_SUMMON_MATURE_BONE_SIFTER), true);
```

File: src/server/scripts/Spells/spell_quest.cpp (hunk 2 above)
Comment (Faq) [reply src/server/scripts/Spells/spell_quest.cpp:2677]:
```suggestion
        return ValidateSpellInfo({ SPELL_SUMMON_SAND_GNOME,SPELL_SUMMON_SAND_GNOME3, SPELL_SUMMON_MATURE_BONE_SIFTER });
```

**Comment by ex1dreamer** [comment]
> I changed the status of this PR to Draft as the SQL bit is missing, so it's clear not to merge it yet.

sorry, pressed wrong button (not ready for review). SQL was already posted by other users. Shall I include? I assumed it was already added.

**Comment by jackpoz** [comment]
> * update SQL table spell_script_names -> spell D=39246, script name=spell_q10930_big_bone_worm

if this requires some SQL statements to be executed, then please include it in this PR

**Comment by offl** [comment]
I'll take a look
In retail aura effect has nothing to do with summoning. Moreover it will be always interrupted in retail by knockback used by spawned creatures
SetDest is not needed here

And first spell summons only 1 gnome or only 1 worm, so old script is correct

The real problem here is for some reason visual effect dissapears(those drums) but that may be not related to scripts

**Comment by offl** [comment]
Idk why you closed this

> SetDest is not needed here
> The real problem here is for some reason visual effect dissapears(those drums) but that may be not related to scripts

Nope, it's needed and fixes that visual issue
My bad, I messed up things

**Pull request closed without merging**